src/regular/jsonrepair.ts

```diff
@@ -519,6 +519,63 @@ export function jsonrepair(text: string): string {
           ) {
             // The quote is followed by the end of the text, a delimiter,
             // or a next value. So the quote is indeed the end of the string.
+
+            // But wait: if followed by a comma, we need to check what comes after.
+            // If what comes after the comma is NOT a valid next value or key,
+            // then this quote is likely an unescaped quote inside the string.
+            if (text[i] === ',') {
+              const iAfterComma = i + 1
+              // skip whitespace after the comma
+              let j = iAfterComma
+              while (j < text.length && isWhitespace(text, j)) {
+                j++
+              }
+
+              // Check if what follows looks like a valid JSON structure continuation
+              const charAfterComma = text[j]
+              if (
+                charAfterComma !== undefined &&
+                charAfterComma !== '}' &&
+                charAfterComma !== ']' &&
+                !isQuote(charAfterComma) &&
+                !isDigit(charAfterComma) &&
+                charAfterComma !== '-' &&
+                charAfterComma !== 't' && // true
+                charAfterComma !== 'f' && // false
+                charAfterComma !== 'n' && // null
+                charAfterComma !== '{' &&
+                charAfterComma !== '['
+              ) {
+                // The content after the comma doesn't look like a valid JSON value.
+                // This means the quote is likely an unescaped quote inside the string.
+                // Check further: if this appears to be followed by more text that ends
+                // the string properly, we should escape this quote.
+                // Look ahead to find if there's a proper ending pattern.
+                let k = j
+                while (k < text.length && !isQuote(text[k]) && text[k] !== '}' && text[k] !== ']') {
```

Comment:
It feels like this while loop and the following if an while loop duplicates logic that is already there: what we try to do here is "parse the next part of the text as if the quote we found is _not_ valid". Can we reuse the existing logic to just ignore the quote at hand and then see if we can still find a valid end quote?